packages/@aws-cdk/aws-route53/lib/records/basic.ts

```diff
@@ -0,0 +1,365 @@
+import { Construct, Resource } from '@aws-cdk/cdk';
+import { IHostedZone } from '../hosted-zone-ref';
+import { CfnRecordSet } from '../route53.generated';
+import { determineFullyQualifiedDomainName } from './_util';
+
+/**
+ * The record type.
+ */
+export enum RecordType {
+  A = 'A',
+  AAAA = 'AAAA',
+  CAA = 'CAA',
+  CNAME = 'CNAME',
+  MX = 'MX',
+  NAPTR = 'NAPTR',
+  NS = 'NS',
+  PTR = 'PTR',
+  SOA = 'SOA',
+  SPF = 'SPF',
+  SRV = 'SRV',
+  TXT = 'TXT'
+}
+
+/**
+ * Propreties for a record.
+ */
+export interface RecordProps {
+  /**
+   * The hosted zone in which to define the new record.
+   */
+  readonly zone: IHostedZone;
+
+  /**
+   * The domain name for this record.
+   *
+   * @default zone root
+   */
+  readonly recordName?: string;
+
+  /**
+   * The resource record cache time to live (TTL) in seconds.
+   *
+   * @default 1800 seconds
+   */
+  readonly ttl?: number;
+}
+
+/**
+ * Construction properties for a BasicRecord.
+ */
+export interface BasicRecordProps extends RecordProps {
+  /**
+   * The record type.
+   */
+  readonly recordType: RecordType;
+
+  /**
+   * The values for this record.
+   */
+  readonly recordValues: string[];
+}
+
+/**
+ * A basic record
+ *
+ * @resource AWS::Route53::RecordSet
+ *
+ * @see https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-values-basic.html
+ */
+export class BasicRecord extends Resource {
```

Comment:
`abstract class`?

Reply:
If not abstract, then  it should probably be called `RecordSet`.

Reply:
I want to highlight the fact that it's a construct for Basic Records only (not alias, failover, geolocation, etc. see https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-values.html).

I don't think it should be abstract, one could use this class to create a record set for record type that is not implemented in an inherited class

Reply:
Sure. If it mirrors the L1 resource it should have the same name I feel. Shouldn't it be called RecordSet then?

Reply:
You could say the same for `AliasRecord` looking at its implementation... 🤔 

I let you decide...

Reply:
Yes, the current design is probably not great. `AliasRecord` shouldn't be a class by itself, but probably `ARecord` needs an `alias` property. 

Do you have any specific changes in mind you would like to propose?

Reply:
You are right, the best solution here would be a base `RecordSet` class with `aliasTarget` moved to **both** `ARecord` and `AaaaRecord`.

Will update accordingly if you're OK with this (requires small changes in `aws-ecs-patterns`, and `aws-route53-targets` also)

Reply:
We then need a union type in `ARecord` right? Whether there are literal targets or  an `aliasTarget`? We do union classes like so these days:

https://github.com/awslabs/aws-cdk/blob/master/packages/@aws-cdk/aws-stepfunctions/lib/input.ts

Reply:
Something like that?
```ts
new ARecord(this, 'IPs', {
  zone: myZone,
  target: RecordSetTarget.fromIpAddresses('1.2.3.4', '5.6.7.8') // (...ipAddresses: string[])
});
```

```ts
new ARecord(this, 'IPs', {
  zone: myZone,
  target: RecordSetTarget.fromAlias(new targets.LoadBalancerTarget(myAlb))
});
```

The `ARecord` would then call `super()` with either `recordValues` or `aliasTarget`.

Another solution is to have both `ipAddresses` and `target` props with a runtime check.

Reply:
Yes, that's what I was thinking. The union type is more in line with what we're currently doing.